mobile/lib/repositories/album.repository.dart

```diff
@@ -118,4 +120,36 @@ class AlbumRepository extends DatabaseRepository implements IAlbumRepository {
   @override
   Future<void> deleteAllLocal() =>
       txn(() => db.albums.where().localIdIsNotNull().deleteAll());
+
+  @override
+  Future<List<Album>> search(
+      String searchTerm, QuickFilterMode filterMode) async {
+    switch (filterMode) {
+      case QuickFilterMode.all:
+        return await db.albums
+            .filter()
```

Comment:
You can simplify these cases by factoring out common parts (like filter, find all) and just modify the query in the switch cases